### PR TITLE
[clang-tidy] Use upper case letters for bool conversion suffix

### PR DESCRIPTION
**Pull request by da-viper**
When readability-implicit-bool-conversion-check and readability-uppercase-literal-suffix-check is enabled this will cause you to apply a fix twice

from (!i) -> (i == 0u) to (i == 0U) twice instead will skip the middle one

For some reason GitHub decided to close the original pull request when I moved the commits to another branch. 
Original pullrequest #102831

### CODE REVIEW

**Review by carlosgalvezp — APPROVED** [review]
LGTM, maybe give a couple more days for other reviewers in case they have more comments.

File: clang-tools-extra/clang-tidy/readability/ImplicitBoolConversionCheck.cpp [hunk 1]
```diff
@@ -392,8 +402,16 @@ void ImplicitBoolConversionCheck::handleCastFromBool(
...
+
+    const auto EquivalentForBoolLiteral =
+        getEquivalentForBoolLiteral(BoolLiteral, DestType, Context);
+    if (UseUpperCaseLiteralSuffix) {
+      Diag << tooling::fixit::createReplacement(
+          *Cast, EquivalentForBoolLiteral.upper());
+    } else {
+      Diag << tooling::fixit::createReplacement(*Cast,
+                                                EquivalentForBoolLiteral);
+    }
```
Comment (5chmidti) [inline clang-tools-extra/clang-tidy/readability/ImplicitBoolConversionCheck.cpp:414]: nit: do not use compound statements for single statements in control structures

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 2]
```diff
@@ -108,6 +108,11 @@ Changes in existing checks
   <clang-tidy/checks/readability/redundant-smartptr-get>` check to
   remove `->`, when reduntant `get()` is removed.
 
+- Improved :doc:`readablility-implicit-bool-conversion
```
Comment (5chmidti) [inline clang-tools-extra/docs/ReleaseNotes.rst:111]: Please sort your entry before `readability-redundant-smartptr-get`

**Review by 5chmidti — APPROVED** [review]
LGTM but with two nits

**Review by carlosgalvezp — APPROVED** [review]
LGTM

**Comment by carlosgalvezp** [comment]
@Da-Viper I believe there's some merge conflicts to fix before being able to merge :)

**Comment by Da-Viper** [comment]
> @Da-Viper I believe there's some merge conflicts to fix before being able to merge :)

rebased

**Comment by carlosgalvezp** [comment]
Thanks, merged! I've edited the commit message a bit to better reflect the content of the patch, as well as mark a related issue as fixed.

**Pull request merged**